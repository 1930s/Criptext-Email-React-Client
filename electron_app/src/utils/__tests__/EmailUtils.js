/* eslint-env node, jest */
const {
  filterCriptextRecipients,
  formOutgoingEmailFromData,
  getRecipientIdFromEmailAddressTag
} = require('./../EmailUtils');
const { appDomain } = require('./../const');

describe('[Form outgoing email] ', () => {
  it('Should form outgoing email from data', () => {
    const labelId = 6;
    const composerData = {
      toEmails: [`toUser@${appDomain}`],
      ccEmails: [`ccUser@${appDomain}`],
      bccEmails: [`bccUser@${appDomain}`],
      secure: true,
      textSubject: 'Subject',
      body: '<p>Hello</p>',
      files: [],
      labelId
    };

    const outgoingData = formOutgoingEmailFromData(composerData);
    outgoingData.emailData.email.date = 1524861748481;
    outgoingData.emailData.email.key = 1524861748481;
    expect(outgoingData).toMatchSnapshot();
  });
});

describe('[Filter emails by domain] ', () => {
  it('Should filter emails by a different domain defined', () => {
    const recipients = [
      `userA@${appDomain}`,
      `userB@${appDomain}`,
      'userC@xxx.com'
    ];
    const filtered = filterCriptextRecipients(recipients);
    const result = [`userA@${appDomain}`, `userB@${appDomain}`];
    expect(filtered).toEqual(result);
  });
});

describe('[Get recipientId from EmailAddressTag] ', () => {
  it('Should get recipientId from criptext domain', () => {
    const from = `"Alice$$" <alice@${appDomain}>`;
    const { recipientId, isExternal } = getRecipientIdFromEmailAddressTag(from);
    const result = 'alice';
    expect(recipientId).toEqual(result);
    expect(isExternal).toBeFalsy();
  });

  it('Should get recipientId from external domain', () => {
    const from = 'Bob 8989 <bob@domain.com>';
    const { recipientId, isExternal } = getRecipientIdFromEmailAddressTag(from);
    const result = 'bob@domain.com';
    expect(recipientId).toEqual(result);
    expect(isExternal).toBeTruthy();
  });

  it('Should get recipientId from external domain, with name comma added', () => {
    const from = `Lola, <lola@domain.com>`;
    const { recipientId, isExternal } = getRecipientIdFromEmailAddressTag(from);
    const result = 'lola@domain.com';
    expect(recipientId).toEqual(result);
    expect(isExternal).toBeTruthy();
  });

  it('Should get recipientId from external domain, with name as emailadresstag', () => {
    const from = `<alice@${appDomain}> <lola@domain.com>`;
    const { recipientId, isExternal } = getRecipientIdFromEmailAddressTag(from);
    const result = 'lola@domain.com';
    expect(recipientId).toEqual(result);
    expect(isExternal).toBeTruthy();
  });

  it('Should get recipientId from external domain, with emailadresstag without tag', () => {
    const from = `alice@domain.com`;
    const { recipientId, isExternal } = getRecipientIdFromEmailAddressTag(from);
    const result = 'alice@domain.com';
    expect(recipientId).toEqual(result);
    expect(isExternal).toBeTruthy();
  });
});
