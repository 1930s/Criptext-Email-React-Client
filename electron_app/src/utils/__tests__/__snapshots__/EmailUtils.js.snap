// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form outgoing email]  Should form outgoing email from data 1`] = `
Object {
  "body": "<p>Hello</p><br/><span style=\\"font-size: 12px;\\">Sent with <a style=\\"color: #0091ff; text-decoration: none;\\" href=\\"https://www.criptext.com/dl\\">Criptext</a></span>",
  "criptextRecipients": Array [
    Object {
      "recipientId": "toUser",
      "type": "to",
    },
    Object {
      "recipientId": "ccUser",
      "type": "cc",
    },
    Object {
      "recipientId": "bccUser",
      "type": "bcc",
    },
  ],
  "emailData": Object {
    "email": Object {
      "content": "<p>Hello</p><br/><span style=\\"font-size: 12px;\\">Sent with <a style=\\"color: #0091ff; text-decoration: none;\\" href=\\"https://www.criptext.com/dl\\">Criptext</a></span>",
      "date": 1524861748481,
      "isMuted": false,
      "key": 1524861748481,
      "preview": "Hello",
      "secure": true,
      "status": 4,
      "subject": "Subject",
      "threadId": undefined,
      "unread": false,
    },
    "fileKeyParams": null,
    "labels": Array [
      6,
    ],
    "recipients": Object {
      "bcc": Array [
        "bccUser@criptext.com",
      ],
      "cc": Array [
        "ccUser@criptext.com",
      ],
      "from": Array [
        "undefined@criptext.com",
      ],
      "to": Array [
        "toUser@criptext.com",
      ],
    },
  },
  "externalRecipients": Object {
    "bcc": Array [],
    "cc": Array [],
    "to": Array [],
  },
}
`;
